flintrock/flintrock.py

```diff
@@ -194,6 +194,10 @@ def cli(cli_context, config, provider):
               help="Git repository to clone Spark from.",
               default='https://github.com/apache/spark',
               show_default=True)
+@click.option('--spark-download-source',
```

Comment:
I suggest moving this option to right after `--spark-version` and updating the help text to read: "URL to download release version of Spark from."

That should make it slightly clearer to the user that these two options are closely related, while the two `--spark-git-...` options are separate.